decadog/src/command/sprint.rs

```diff
@@ -234,21 +238,38 @@ fn finish_sprint(settings: &Settings) -> Result<(), Error> {
         };
 
         println!("{}", &issue);
+        // This bool keeps track of whether an issue was planned or not. Issues are considered
+        // planned if they belong to the current milestone at time of closing the sprint. If an
+        // issue is added to the milestone at the end of the sprint, then it is considered as out
+        // of sprint.
+        issue_is_planned = true;
 
-        // If no milestone, ask to assign to open milestone
+        // If no milestone, ask to assign to open milestone and if applicable mark as not planned
         // If answer is no, ignore this issue
         if issue.milestone.is_none() {
             if Confirmation::new("Assign to milestone?").interact()? {
                 client.assign_issue_to_milestone(&issue, Some(&open_milestone))?;
```

Comment:
If we were writing a production-grade service (hint: we're not) I would batch up all the `assign_issue_to_milestone` calls and execute them at the end, once the user has confirmed everything. You might like to open a new ticket for this though?

The reason is, at the moment, if the user enters an invalid number of points that were planned in the sprint, we crash and start again. But we've already done these milestone assignment calls, so we now can't determine what was planned and what wasn't when we retry closing the sprint.

It would be best to not lose information (milestone assignment, title change) until we've completed all other user input/IO, and then get it all done in one go.

Reply:
Good call. Let's discuss this.